core/server/version.go

```diff
@@ -15,10 +21,66 @@ var (
 )
 
 func (cs *coreServer) GetVersion(ctx context.Context, msg *pb.GetVersionRequest) (*pb.GetVersionResponse, error) {
+	u := &unstructured.Unstructured{}
+
+	u.SetGroupVersionKind(schema.GroupVersionKind{
+		Group:   "",
+		Version: "v1",
+		Kind:    "Namespace",
+	})
+
+	var ns string
+
+	if msg.Namespace != "" {
+		ns = msg.Namespace
+	} else {
+		ns = wego.DefaultNamespace
+	}
+
+	key := client.ObjectKey{
+		Name: ns,
+	}
+
+	fluxVersion := ""
+
+	clustersClient, err := cs.clientsFactory.GetImpersonatedClient(ctx, auth.Principal(ctx))
+	if err != nil {
+		cs.logger.Error(err, "error getting impersonating client")
+	} else {
+		c, err := clustersClient.Scoped(clustersmngr.DefaultCluster)
+		if err != nil {
+			cs.logger.Error(err, "error getting scoped client")
+		} else {
+			err = c.Get(ctx, key, u)
+			if err != nil {
+				cs.logger.Error(err, "error getting object")
+			} else {
+				fluxVersion = u.GetLabels()["app.kubernetes.io/version"]
+			}
+		}
+	}
```

Comment:
@ozamosi @chanwit suggestions on how to simplify nested ifs here (looks similar to a "callback hell" in JS) welcome!

I don't want to return nil from `GetVersion` on error when getting additional values, because I want to display at least available values (at least the values set at the build time) in the app for debug purposes.

Reply:
Maybe lift this bit into a new function, return an error from there any time there's an error, and then this function only has one error to check?

Reply:
Thanks, good idea, will do it.